### PR TITLE
Make LossScaleOpimtizer wrapper compatible with decoupled weight decay

### PR DESCRIPTION
**Pull request by ben-arnao**
This allows for us to use decoupled weight decay optimizers with mixed precision.

https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/weight_decay_optimizers/DecoupledWeightDecayExtension

### CODE REVIEW

**Comment by ben-arnao** [comment]
@googlebot I signed it!

**Comment by reedwm** [comment]
Right now, we intentionally do not expose attributes in the wrapped optimizer, so I'm closing this PR However, LossScaleOptimizer should still be able to wrap a DecoupledWeightDecayExtension optimizer. Let me know if there is an incompatibility between LossScaleOptimizer and DecoupledWeightDecayExtension.

**Comment by ben-arnao** [comment]
> LossScaleOptimizer

The problem doesn't arise from wrapping the optimizer directly, but if we need to dynamically change the weight_decay like we do with lr in a callback like ReduceLROnPlateau then we run into problems.

We can get and set the lr in the callback

```
K.get_value(self.model.optimizer.lr)
K.set_value(self.model.optimizer.lr, new_lr)
```

But we cannot do the same for weight decay (which is required when used decoupled weight decay)

```
K.get_value(self.model.optimizer.weight_decay)
K.set_value(self.model.optimizer.weight_decay, new_wd)
```
I believe we should be able to decay lr/wd in the same manner.

**Comment by reedwm** [comment]
We did special case `lr` since almost every optimizer has a learning rate. I agree you should able to set the weight decay as well, but it's not feasible to manually define every attribute that an optimizer may define.

I think the solution here is to either modify `__getattribute__` to delegate arbitrary attributes to the inner optimizer or to create a dynamic class which subclasses both the wrapped optimizer and LossScaleOptimizer. The latter is [what DecoupledWeightDecayExtension does](https://github.com/tensorflow/addons/blob/1af92905ed03f05fcf6f4918783c3d151a8b8350/tensorflow_addons/optimizers/weight_decay_optimizers.py#L183). I will work on implementing one of these and try to get it into TF 2.3.

As a workaround, for now you can access the weight decay with
```
if isinstance(optimizer, tf.keras.mixed_precision.experimental.LossScaleOptimizer):
    weight_decay = optimizer._optimizer.weight_decay
else:
    weight_decay = optimizer.weight_decay
```

Note accessing `optimizer._optimizer` may break in future versions of TensorFlow.
